### ISSUE TITLE
Inconsistent raising of DocumentMissingException between DELETE and UPDATE

### ISSUE THREAD

**Issue by hlcianfagna** [crate/crate#14707]
### CrateDB version

nightly-5.5.0-20230918

### CrateDB setup information

discovery.type=single-node

### Problem description

When a REFRESH has not yet taken place and DELETE identifies for deletion a record that has in the meanwhile already been deleted, DELETE reports success with 0 rows affected
but when UPDATE identifies for update a record that has been deleted it fails with DocumentMissingException

### Steps to Reproduce

```sql
create table tst1 (a int primary key,b int) with (refresh_interval=900000);
insert into tst1 select 1,2;
refresh table tst1;
delete from tst1;
--> DELETE OK, 1 row affected  (0.010 sec)
delete from tst1;
--> DELETE OK, 0 rows affected  (0.009 sec)
update tst1 set b=3;
--> DocumentMissingException[[default][1]: document missing]
```

### Actual Result

See above

### Expected Result

Either `DELETE` also failing with DocumentMissingException or UPDATE succeeding with `0 rows affected`

**Comment by hlcianfagna** [comment]
Relates with https://github.com/crate/crate/issues/14386

It could be considered a documentation bug if the behaviour is by design.

**Comment by seut** [comment]
Looks to me that `UPDATE` should not return an exception in this case, but just `0 Rows` (as it will just not find any documents to update).
Adding a where clause to the UPDATE will behave like this already -> no rows affected.
`update tst1 set b = b where a = 1; `

**Comment by BaurzhanSakhariev** [comment]
Hi @hlcianfagna, thanks for reporting!

Fix will be available in the 5.4.4 release.
